core/arch/riscv/include/kernel/tlb_helpers.h

```diff
@@ -9,7 +9,19 @@
 #ifndef __ASSEMBLER__
 
 void tlbi_all(void);
+void tlbi_va_allasid(vaddr_t va);
 void tlbi_asid(unsigned long asid);
+void tlbi_va_asid(vaddr_t va, uint32_t asid);
+
+static inline void tlbi_va_allasid_nosync(vaddr_t va)
```

Comment:
What's the point of these _nosync functions? Do you plan to provide more primitive versions later that will be more efficient for bulk operations?

Reply:
@jenswi-linaro 
Hmm.. In fact, _nosync are meaningless for RISC-V.

Hi @maroueneboubakri 

Do we need to flush icache after flushing TLB by sfence.vma?
If not, I think we do not need _nosync related functions, what do you think?
In `core_mmu_arch.c`:

```C
void tlbi_va_range(vaddr_t va, size_t len, size_t granule)
{
	assert(granule == CORE_MMU_PGDIR_SIZE || granule == SMALL_PAGE_SIZE);
	assert(!(va & (granule - 1)) && !(len & (granule - 1)));

	mb();
	while (len) {
		tlbi_va_allasid(va);
		len -= granule;
		va += granule;
	}
	mb();
}

void tlbi_va_range_asid(vaddr_t va, size_t len, size_t granule, uint32_t asid)
{
	assert(granule == CORE_MMU_PGDIR_SIZE || granule == SMALL_PAGE_SIZE);
	assert(!(va & (granule - 1)) && !(len & (granule - 1)));

	mb();
	while (len) {
		tlbi_va_asid(va, asid);
		len -= granule;
		va += granule;
	}
	mb();
}
```

Reply:
@gagachang I agree, I remove `_nosync` then.